src/UKHO.S100PermitService.Common/Providers/CacheProvider.cs

```diff
@@ -0,0 +1,28 @@
+﻿using Microsoft.Extensions.Caching.Memory;
+using System.Diagnostics.CodeAnalysis;
+
+namespace UKHO.S100PermitService.Common.Providers
+{
+    [ExcludeFromCodeCoverage]
+    public class CacheProvider : ICacheProvider
+    {
+        private readonly IMemoryCache _memoryCache;
+
+        public CacheProvider(IMemoryCache memoryCache)
+        {
+            _memoryCache = memoryCache;                     
+        }
+
+        public string GetCacheKey(string key)
+        {
+            _memoryCache.TryGetValue(key, out string? Value);
+            return Value;
+        }
+
+        public string SetCacheKey(string key, string value, TimeSpan timeSpan)
+        {            
+            _memoryCache.Set(key, value, timeSpan);            
+            return value;
```

Comment:
Why we are returning the value from set method?

Reply:
Modified, Thanks